### PR TITLE
Store the database in a role specific subdirectory

### PR DESCRIPTION
**Pull request by hirschenberger**
This is a cleaned up version of #8658 fixing #6880

polkadot companion: paritytech/polkadot#2923

### CODE REVIEW

File: client/db/src/utils.rs [hunk 1]
```diff
@@ -394,6 +424,33 @@ pub fn check_database_type(
 	Ok(())
 }
 
+fn maybe_migrate_to_type_subdir(p: &Path, db_type: DatabaseType) -> io::Result<PathBuf> {
```
[the remark this answers is not in the document]
Comment (cheme) [reply client/db/src/utils.rs:427]: I am actually not sure it should be remove in the future, but consider it a good idea.

**Review by cheme — COMMENTED** [review]
Looks good to me.
It needs to be merge with master though for ci to pass.
The migration will not work for parity_db (no 'db_version' file but 'metadata' can be use).

**Comment by hirschenberger** [comment]
Don't know what's the problem with the companion?

**Comment by cheme** [comment]
> Don't know what's the problem with the companion?

Probably need merge with master

**Comment by hirschenberger** [comment]
> > Don't know what's the problem with the companion?
> 
> Probably need merge with master

I just did merge but after the commit here, can you rerun the pipline here?

**Review by cheme — APPROVED** [review]
LGTM

**Review by arkpar — APPROVED** [review]

**Comment by ordian** [comment]
bot merge

**Pull request merged**